public/components/common/seach/queryBar.tsx

```diff
@@ -9,28 +9,83 @@
  * GitHub history for details.
  */
 
-import React from 'react';
+import React, {Fragment, useEffect, useState, useCallback} from 'react';
 import {
+  EuiButtonEmpty,
   EuiFieldSearch,
   EuiFlexItem,
+  EuiSuggest,
+  EuiFlexGroup,
+  EuiSelectableTemplateSitewide,
+  EuiSelectableTemplateSitewideOption,
+  EuiSuggestItem,
+  EuiSuggestItemProps
 } from '@elastic/eui';
+import useDebounce from 'react-use/lib/useDebounce'
```

Comment:
initially, i wanted to follow someone's idea about using Debounce but i couldn't get it to work. will be removing

Reply:
i see, you could try lodash debounce if you need to. example here: https://github.com/opensearch-project/sql/blob/main/workbench/public/components/Main/main.tsx#L241-L242

But i don't think it's necessary to debounce since search is triggered by the enter key, not incremental